### ISSUE TITLE
opts for jobstart is missing stdout/stderr field for channel-buffered mode in lua 

### ISSUE THREAD

**Issue by ddickstein** [neovim/neovim#22940]
`:h channel-buffered` says, "set the `stdout_buffered`, `stderr_buffered`, `stdin_buffered`, or `data_buffered` option keys to invoke the callback only after all output was gathered and the stream was closed. If a buffering mode is used without a callback, the data is saved in the stream {name} key of the options dict." This works correctly in VimL but does not work in Lua. Compare the following:

```viml
function! Viml_example()
  let opts = {}
  let opts.stdout_buffered = v:true
  let opts.stderr_buffered = v:true
  function opts.on_exit(job_id, exit_code, event) closure
    call nvim_out_write(string(opts)."\n")
  endfunction
  call jobstart(['true'], opts)
endfunction
```

```lua
function Lua_example()
  local opts = {}
  opts.stdout_buffered = true
  opts.stderr_buffered = true
  function opts.on_exit()
    vim.api.nvim_out_write(vim.fn.string(opts).."\n")
  end
  vim.fn.jobstart({'true'}, opts)
end
```

```viml
:call Viml_example()
{'stderr': [''], 'stdout_buffered': v:true, 'on_exit': function('1'), 'stderr_buffered': v:true, 'stdout': ['']}
:lua Lua_example()
{'stdout_buffered': v:true, 'on_exit': function('<lambda>2'), 'stderr_buffered': v:true}
```

### Neovim version (nvim -v)

0.9.0-dev-1347+gfd32a9875

### Vim (not Nvim) behaves the same?

N/A

### Operating system/version

Rocky Linux 8.6

### Terminal name/version

xfce4-terminal 1.0.4 

### $TERM environment variable

tmux-256color

### Installation

appimage

**Comment by zeertzjq** [comment]
Looks similar to #22863

**Comment by bfredl** [comment]
likely we want to solve this by lua native wrappers around uv.spawn. something like https://github.com/neovim/neovim/pull/17620

**Comment by zeertzjq** [comment]
Duplicate of #22213
